### PR TITLE
Resolves #565 - Implements 'navigation:homeButton'

### LINKED ISSUE #565: There is no 'navigation:homeButton' event
There is a handler for 'navigation:backButton' which redirects the user to the previous route but there is no such handler for 'navigation:homeButton'.

### PR DESCRIPTION
**Pull request by brian-learningpool**
It makes sense to implement a handler for this, since 'backButton' is
already supported.

### CODE REVIEW

File: src/core/js/router.js [hunk 1]
```diff
@@ -101,6 +102,10 @@ define(function(require) {
             }
         },
 
+        navigateToHomeRoute: function() {
+            this.navigate('#', {trigger: true});
```
Comment (oliverfoster) [inline src/core/js/router.js]: Can we have if (Adapt.router.get('_canNavigate')) {  in there as well?

We've an extension called 'promptOnPageIncomplete'. This'd be quite usefull.

Comment (brian-learningpool) [inline src/core/js/router.js]: Good point.  I've updated the code.

**Comment by oliverfoster** [comment]
+1

**Comment by darylhedley** [comment]
+1

**Comment by hrajotia** [comment]
+1

**Comment by taylortom** [comment]
@brian-learningpool can we get this put into develop too?

**Pull request merged**